fastly/block_fastly_service_v1_logging_openstack.go

```diff
@@ -0,0 +1,289 @@
+		BucketName:        gofastly.NullString(df["bucket_name"].(string)),
+		AccessKey:         gofastly.NullString(df["access_key"].(string)),
+		PublicKey:         gofastly.NullString(df["public_key"].(string)),
+		GzipLevel:         gofastly.Uint(uint(df["gzip_level"].(int))),
+		MessageType:       gofastly.NullString(df["message_type"].(string)),
+		Path:              gofastly.NullString(df["path"].(string)),
+		Period:            gofastly.Uint(uint(df["period"].(int))),
+		TimestampFormat:   gofastly.NullString(df["timestamp_format"].(string)),
+		Format:            gofastly.NullString(df["format"].(string)),
+		FormatVersion:     gofastly.Uint(uint(df["format_version"].(int))),
+		Placement:         gofastly.NullString(df["placement"].(string)),
+		ResponseCondition: gofastly.NullString(df["response_condition"].(string)),
+	}
+}
+
+func buildDeleteOpenstack(openstackMap interface{}, serviceID string, serviceVersion int) *gofastly.DeleteOpenstackInput {
+	df := openstackMap.(map[string]interface{})
+
+	return &gofastly.DeleteOpenstackInput{
+		Service: serviceID,
+		Version: serviceVersion,
+		Name:    df["name"].(string),
+	}
+}
+
+func (h *OpenstackServiceAttributeHandler) Register(s *schema.Resource) error {
+	s.Schema[h.GetKey()] = &schema.Schema{
+		Type:     schema.TypeSet,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				// Required fields
+				"name": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "The unique name of the OpenStack logging endpoint.",
+				},
+
+				"url": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "Your OpenStack auth url.",
+				},
+
+				"user": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "The username for your OpenStack account.",
+				},
+
+				"bucket_name": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "The name of your OpenStack container.",
+				},
+
+				"access_key": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Sensitive:   true,
+					Description: "Your OpenStack account access key.",
+				},
+
+				// Optional fields
+				"public_key": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "A PGP public key that Fastly will use to encrypt your log files before writing them to disk.",
+				},
+
+				"gzip_level": {
+					Type:        schema.TypeInt,
+					Optional:    true,
+					Default:     0,
+					Description: "What level of GZIP encoding to have when dumping logs (default 0, no compression).",
+				},
+
+				"period": {
+					Type:        schema.TypeInt,
+					Optional:    true,
+					Default:     3600,
+					Description: "How frequently log files are finalized so they can be available for reading (in seconds, default 3600).",
+				},
+
+				"path": {
+					Type:        schema.TypeString,
+					Optional:    true,
```

Comment:
Maybe we should set a default value of `"/"`?

We don't set a default for `S3` - https://github.com/terraform-providers/terraform-provider-fastly/blob/master/fastly/block_fastly_service_v1_s3logging.go#L112

`manage.fastly.com` sets the default to `"/"`, however, this is not set by the backend API.

Reply:
From a quick grep, the backend doesn't set a default for path for any logging endpoint. We do have it noted that we should have the backend set `/` as the default when none is set by the client for logging endpoints where it would make sense to do so.